solidity/contracts/token/extensions/TokenBridgeCctp.sol

```diff
@@ -0,0 +1,138 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+import {TypeCasts} from "../../libs/TypeCasts.sol";
+import {TokenRouter} from "../libs/TokenRouter.sol";
+import {TokenMessage} from "../libs/TokenMessage.sol";
+import {HypERC20Collateral} from "../HypERC20Collateral.sol";
+import {Quote, ITokenBridge} from "../../interfaces/ITokenBridge.sol";
+import {ITokenMessenger} from "../../interfaces/cctp/ITokenMessenger.sol";
+import {ITokenMessengerV2} from "../../interfaces/cctp/ITokenMessengerV2.sol";
+import {IMessageTransmitter} from "../../interfaces/cctp/IMessageTransmitter.sol";
+import {StandardHookMetadata} from "../../hooks/libs/StandardHookMetadata.sol";
+
+contract TokenBridgeCctp is ITokenBridge, HypERC20Collateral {
+    using TypeCasts for bytes32;
+
+    uint32 internal constant CCTP_VERSION_1 = 0;
+    uint32 internal constant CCTP_VERSION_2 = 1;
+
+    // @dev the minimum to consider it a Standard CCTP transfer (it applies to every network)
+    // see https://github.com/circlefin/evm-cctp-contracts/blob/release-2025-03-11T143015/src/v2/MessageTransmitterV2.sol#L224-L244
+    // and https://github.com/circlefin/evm-cctp-contracts/blob/release-2025-03-11T143015/src/v2/FinalityThresholds.sol#L21
+    uint32 internal constant CCTP_V2_DEFAULT_MIN_FINALITY_THRESHOLD = 2000;
+
+    // @dev required for CCTP v2
+    uint256 internal constant CCTP_V2_DEFAULT_MAX_FEE = 0;
+
+    // CCTP token messenger
+    // NOTE: be sure to use TokenMessenger (and MessageTransmitter on the ISM)
+    // that support the same CCTP version
+    ITokenMessenger immutable tokenMessenger;
```

Comment:
we can guarantee this if we deploy the ISM internally to this contract's constructor

Reply:
I thought I had previously left this comment already, but imo `TokenBridgeCctp.sol` should just be the ISM itself? I don't see the reason why its a separate contract and would make artifact management easier

Reply:
yeah, I'm finalizing these changes right now along with an E2E test, should push everything shortly